bin/schema_generator.sh

```diff
@@ -0,0 +1,156 @@
+# mozilla-pipeline-schemas.
+#
+# Environment variables:
+#   MPS_SSH_KEY_BASE64: A base64-encoded ssh secret key with permissions to push
+#                       to mozilla-pipeline-schemas
+#
+# Example usage:
+#   export MPS_SSH_KEY_BASE64=$(cat ~/.ssh/id_rsa | base64)
+#   make build && make run
+#
+# TODO: Update schema mapping for validation
+# TODO: Handle overwriting glean schemas
+# TODO: Include Main Ping from schema generation
+# TODO: What the heck to do with pioneer-study, a non-nested namespace
+
+MPS_BRANCH_SOURCE="dev"
+MPS_BRANCH_WORKING="generated-schemas-dev"
+MPS_BRANCH_PUBLISH="generated-schemas"
+MPS_SCHEMAS_DIR="schemas"
+
+BASE_DIR="/app"
+
+
+function setup_git_ssh() {
+    # Configure the container for pushing to github
+
+    if [[ -z "$MPS_SSH_KEY_BASE64" ]]; then
+        echo "Missing secret key" 1>&2
+        exit 1
+    fi
+
+    git config --global user.name "Generated Schema Creator"
+    git config --global user.email "dataops+pipeline-schemas@mozilla.com"
+
+    mkdir -p /app/.ssh
+
+    echo "$MPS_SSH_KEY_BASE64" | base64 --decode > /app/.ssh/id_ed25519
+    ssh-keyscan github.com > /app/.ssh/known_hosts # Makes the future git-push non-interactive
+
+    chown -R "$(id -u):$(id -g)" "$HOME/.ssh"
+    chmod 700 "$HOME/.ssh"
+    chmod 700 "$HOME/.ssh/id_ed25519"
+}
+
+function setup_dependencies() {
+    # Installs mozilla-schema-generator in a virtual environment
+
+    virtualenv msg-venv
+    # shellcheck disable=SC1091
+    source msg-venv/bin/activate
+    pip install -e ./mozilla-schema-generator
+}
+
+function clone_and_configure_mps() {
+    # Checkout mozilla-pipeline-schemas and changes directory to prepare for
+    # schema generation.
+
+    rm -rf mozilla-pipeline-schemas
+
+    git clone git@github.com:mozilla-services/mozilla-pipeline-schemas.git
+    cd mozilla-pipeline-schemas/$MPS_SCHEMAS_DIR || exit
+    git checkout $MPS_BRANCH_SOURCE
+
+    git branch -D $MPS_BRANCH_WORKING
+    git checkout -b $MPS_BRANCH_WORKING
+}
+
+function prepare_metadata() {
+    # Assumes that all schemas under the current directory are valid JSON schema.
+
+    local telemetry_metadata="metadata/telemetry-ingestion/telemetry-ingestion.1.schema.json"
+    local structured_metadata="metadata/structured-ingestion/structured-ingestion.1.schema.json"
+
+    find ./telemetry -type f -exec metadata_merge $telemetry_metadata {} ";"
+    find . -path ./telemetry -prune -o -type f -exec metadata_merge $structured_metadata {} ";"
+}
+
+function filter_schemas() {
+    # Remove metadata
+    rm -rf metadata
+
+    # Pioneer-study is not nested, remove it
+    rm -rf pioneer-study
+
+    # Replace newlines with backticks (hard to do with sed): cat | tr
+    # Remove the last backtick; it's the file-ending newline: rev | cut | rev
+    # Replace backticks with "\|" (can't do that with tr): sed
+    # Find directories that don't match any of the regex expressions: find
```

Comment:
The goal of this step is to remove bq schemas not in `allowlist`, we are keeping the directories and json schemas.